work.go

```diff
@@ -103,15 +151,16 @@ func (w *worker) addErrors(throttle <-chan interface{}, limit, framesMin, frames
 		return w
 	}
 	w.Add(func(done <-chan struct{}) error {
-		for cnt := 0; cnt < limit; cnt++ {
+		var sent int
+		for sent < limit {
```

Comment:
just a nitpick - these are really generated not sent.  I hope we never need to make a distinction as that would indicate a problem with the agent.